ayon_server/addons/rezbuild.py

```diff
@@ -0,0 +1,112 @@
+""" AYON Rez Builder
+
+This expects a certain folder structure:
+├── client
+│   └── <client module name>  # Example: ayon_core
+├── package.py
+└── server
+
+Other folders might or might not be present, but they are essentially ignored.
+
+Will zip up the `client/<client module name>` and place it into `private/client.zip`
+and copy the contents of `server` into the root of the package.
+
+<package root>
+├── frontend
+│   ├── dist
+├── server
+│   ├── __init__.py
+│   └── settings
+├── private
+│   ├── client.zip
+│   └── pyproject.toml
+├── package.py
+└── build.rtx
+
+MIght be worth checking in the future https://gitlab.com/Pili-Pala/rezbuild
+"""
+import os
+from pathlib import Path
+import shutil
+
+SRC_DIR = Path(os.environ["REZ_BUILD_SOURCE_PATH"])
+DEST_DIR = Path(os.environ["REZ_BUILD_INSTALL_PATH"])
+
+
+def _recursive_copy(src, dst):
+    """ Recursively travers a directory and copy the contents to destination
+
+    For soem reason `shutil.copytree` refuses to work, so I had to implement.
+    """
+    for child in src.iterdir():
+        if child.is_dir():
+            _recursive_copy(child, dst / child.stem)
+        elif child.is_file():
+            dst_file = dst / child.relative_to(src)
+            print(f"Copying {child} -> {dst_file}")
+            dst_file.parent.mkdir(parents=True, exist_ok=True)
+            shutil.copy(
+                child,
+                dst_file
+            )
+
+
+def install_addon():
+    server_dir = SRC_DIR / "server"
+
+    # Copy "server" to "addon"
+    if server_dir.is_dir():
+        print(f"Copying `server` directory contents to {DEST_DIR}")
+        _recursive_copy(server_dir, DEST_DIR)
```

Comment:
Not what this does, but if there is `package.py` it does expect to have server code in `server` subfolder.
```suggestion
        _recursive_copy(server_dir, DEST_DIR / "server")
```